client/client.go

```diff
@@ -33,20 +38,131 @@ import (
 	"github.com/transparency-dev/trillian-tessera/api"
 	"github.com/transparency-dev/trillian-tessera/api/layout"
 	"golang.org/x/mod/sumdb/note"
+	"k8s.io/klog/v2"
 )
 
 var (
 	hasher = rfc6962.DefaultHasher
 )
 
-// Fetcher is the signature of a function which can retrieve arbitrary files from
-// a log's data storage, via whatever appropriate mechanism.
-// The path parameter is relative to the root of the log storage.
+// CheckpointFetcherFunc is the signature of a function which can retrieve the latest
+// checkpoint from a log's data storage.
 //
 // Note that the implementation of this MUST return (either directly or wrapped)
 // an os.ErrIsNotExit when the file referenced by path does not exist, e.g. a HTTP
 // based implementation MUST return this error when it receives a 404 StatusCode.
-type Fetcher func(ctx context.Context, path string) ([]byte, error)
+type CheckpointFetcherFunc func(ctx context.Context) ([]byte, error)
+
+// TileFetcherFunc is the signature of a function which can fetch the raw data
+// for a given tile.
+//
+// Note that the implementation of this MUST return (either directly or wrapped)
+// an os.ErrIsNotExit when the file referenced by path does not exist, e.g. a HTTP
+// based implementation MUST return this error when it receives a 404 StatusCode.
+type TileFetcherFunc func(ctx context.Context, level, index, logSize uint64) ([]byte, error)
+
+// EntryBundleFetcherFunc is the signature of a function which can fetch the raw data
+// for a given entry bundle.
+//
+// Note that the implementation of this MUST return (either directly or wrapped)
+// an os.ErrIsNotExit when the file referenced by path does not exist, e.g. a HTTP
+// based implementation MUST return this error when it receives a 404 StatusCode.
+type EntryBundleFetcherFunc func(ctx context.Context, bundleIndex, logSize uint64) ([]byte, error)
+
+// NewHTTPFetcher creates a new HTTPFetcher for the log rooted at the given URL, using
+// the provided HTTP client.
+//
+// rootURL should end in a trailing slash.
+// c may be nil, in which case http.DefaultClient will be used.
+func NewHTTPFetcher(rootURL *url.URL, c *http.Client) (*HTTPFetcher, error) {
```

Comment:
What do you think about putting all this fetcher code in a fetcher.go file in this same package?

Reply:
I think that's a good idea!
done.